cloudinit/net/networkd.py

```diff
@@ -0,0 +1,225 @@
+            elif k == 'metric':
+                cfg.update_section(sec, 'Metric', v)
+
+    def parse_subnets(self, iface, cfg):
+        dhcp = 'no'
+        for e in iface.get('subnets', []):
+            t = e['type']
+            if t == 'dhcp4' or t == 'dhcp':
+                if dhcp == 'no':
+                    dhcp = 'ipv4'
+                elif dhcp == 'ipv6':
+                    dhcp = 'yes'
+            elif t == 'dhcp6':
+                if dhcp == 'no':
+                    dhcp = 'ipv6'
+                elif dhcp == 'ipv4':
+                    dhcp = 'yes'
+            if 'routes' in e and e['routes']:
+                for i in e['routes']:
+                    self.parse_routes(i, cfg)
+            elif 'address' in e:
+                for k, v in e.items():
+                    if k == 'address':
+                        tmp = v
+                        if 'prefix' in e:
+                            tmp += '/' + str(e['prefix'])
+                        cfg.update_section('Address', 'Address', tmp)
+                    elif k == 'gateway':
+                        cfg.update_section('Route', 'Gateway', v)
+                    elif k == 'dns_nameservers':
+                        cfg.update_section('Network', 'DNS', ' '.join(v))
+                    elif k == 'dns_search':
+                        cfg.update_section('Network', 'Domains', ' '.join(v))
+
+        cfg.update_section('Network', 'DHCP', dhcp)
+
+    def dhcp_domain(self, d, cfg):
+        for item in ['dhcp4domain', 'dhcp6domain']:
+            if item not in d:
+                continue
+            ret = str(d[item]).casefold()
+            try:
+                ret = util.translate_bool(ret)
+                ret = 'yes' if ret else 'no'
+            except ValueError:
+                if ret != 'route':
+                    LOG.warning('Invalid dhcp4domain value - %s', ret)
+                    ret = 'no'
+            if item == 'dhcp4domain':
+                section = 'DHCPv4'
+            else:
+                section = 'DHCPv6'
+            cfg.update_section(section, 'UseDomains', ret)
+
+    def parse_dns(self, iface, cfg):
+        sec = 'Network'
+        dns = iface.get('dns')
+
+        if not dns:
+            return
+
+        if 'search' in dns and dns['search']:
+            cfg.update_section(sec, 'Domains', ' '.join(dns['search']))
+
+        if 'nameservers' in dns and dns['nameservers']:
+            cfg.update_section(sec, 'DNS', ' '.join(dns['nameservers']))
+
+    def create_network_file(self, link, conf):
+        LOG.debug('Setting Networking Config for %s', link)
+
+        net_fn = self.network_fname_fptr(self.network_conf_dir, link)
+        util.write_file(net_fn, conf)
+        util.chownbyname(net_fn, 'systemd-network', 'systemd-network')
+
+    def render_network_state(self, network_state, templates=None, target=None):
+        fpnplan = os.path.join(subp.target_path(target), self.network_conf_dir)
```

Comment:
This needs to be used when creating the network file. This is especially important when using the `net-convert` command.

E.g., if I run:
`cloud-init devel net-convert -p /tmp/test.yaml -k yaml -d /tmp -D ubuntu -O networkd --debug`

I'm saying with the `-d tmp` that  I want the renderer to write the files to `/tmp/etc/systemd/network`. As it is now, it'll write to `/etc/systemd/network`, which isn't what we want.